dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java

```diff
@@ -679,64 +681,87 @@ private void deleteTreesForInodes(List<String> inodes) throws DotDataException {
-
-        int deleted=before - after;
-
-        // deleting orphan binary files
-        java.io.File assets=new java.io.File(APILocator.getFileAssetAPI().getRealAssetsRootPath());
-        for(java.io.File ff1 : assets.listFiles())
-            if(ff1.isDirectory() && ff1.getName().length()==1 && ff1.getName().matches("^[a-f0-9]$"))
-                for(java.io.File ff2 : ff1.listFiles())
-                    if(ff2.isDirectory() && ff2.getName().length()==1 && ff2.getName().matches("^[a-f0-9]$"))
-                        for(java.io.File ff3 : ff2.listFiles())
-                            try {
-                                if(ff3.isDirectory()) {
-                                    Contentlet con=find(ff3.getName());
-                                    if(con==null || !UtilMethods.isSet(con.getIdentifier()))
-                                        if(!FileUtils.deleteQuietly(ff3))
-                                            Logger.warn(this, "can't delete "+ff3.getAbsolutePath());
+        final int after = Integer.parseInt(result.get(0).get("count"));
+        final int deleted = before - after;
+        if (deleted > 0) {
+            // Deleting orphaned binary files
+            final java.io.File assets = new java.io.File(APILocator.getFileAssetAPI().getRealAssetsRootPath());
+            for (final java.io.File firstLevelFolder : assets.listFiles()) {
+                if (firstLevelFolder.isDirectory() && firstLevelFolder.getName().length() == 1 && firstLevelFolder
+                        .getName().matches("^[a-f0-9]$")) {
+                    for (final java.io.File secondLevelFolder : firstLevelFolder.listFiles()) {
+                        if (secondLevelFolder.isDirectory() && secondLevelFolder.getName().length() == 1 &&
+                                secondLevelFolder.getName().matches("^[a-f0-9]$")) {
+                            for (final java.io.File asset : secondLevelFolder.listFiles()) {
+                                try {
+                                    if (asset.isDirectory()) {
+                                        final Contentlet contentlet = find(asset.getName());
+                                        if (null == contentlet || !UtilMethods.isSet(contentlet.getIdentifier())) {
+                                            if (!FileUtils.deleteQuietly(asset)) {
+                                                Logger.warn(this, "Asset '" + asset.getAbsolutePath() + "' could " +
+                                                        "not be deleted.");
+                                            }
+                                        }
+                                    }
+                                } catch (final Exception ex) {
+                                    Logger.warn(this, String.format("An error occurred when deleting asset '%s': %s",
+                                            asset.getAbsolutePath(), ex.getMessage()));
                                 }
```

Comment:
All this logic should go into a method too

https://github.com/dotCMS/core/pull/18507/files#diff-0d54bf61089532e67d512211809b7a69R731-R755

Reply:
Fixed.